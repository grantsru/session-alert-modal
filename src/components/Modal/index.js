import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Dialog from "@material-ui/core/Dialog";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";

const styles = () => ({
  title: {
    borderBottom: "1px solid #CCC",
    textAlign: "center"
  },
  content: {
    padding: "25px",
    minHeight: "45px",
    textAlign: "center"
  }
});

const Modal = ({ classes, content, fullScreen, loading, open, title }) => (
  <Dialog
    open={open}
    transitionDuration={1500}
    fullScreen={fullScreen}
    maxWidth={"xs"}
  >
    <DialogTitle className={classes.title}>
      {title}
    </DialogTitle>
    <React.Fragment>
      <DialogContent className={classes.content}>
      {content}
      </DialogContent>
    </React.Fragment>
  </Dialog>
);

Modal.propTypes = {
  classes: PropTypes.object.isRequired,
  content: PropTypes.node.isRequired,
  loading: PropTypes.bool.isRequired,
  open: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired
};

export default withStyles(styles)(Modal);
