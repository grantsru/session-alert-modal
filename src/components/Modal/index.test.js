import React from "react";
import Modal from "./index";
import { create } from "react-test-renderer";

const login = () => null;
const handleButtonClick = () => null;

describe("Modal", () => {
  test("it matches the snapshot", () => {
    const component = create(
      <Modal
        content={<React.Fragment>Foo</React.Fragment>}
        loading={false}
        open={false}
        title="Foo"
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
