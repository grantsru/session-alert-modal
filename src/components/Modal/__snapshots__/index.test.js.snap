// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal it matches the snapshot 1`] = `null`;
